server.go

```diff
@@ -222,6 +215,9 @@ func (server *Server) isAbleToEndSession(message *Message, session sessionInterf
 func (server *Server) handleSession(session sessionInterface) {
 	defer session.finish()
 	message, configuration := server.newMessage(), server.configuration
+	defer func() {
+		server.messages.append(message)
```

Comment:
And we append it at the end of the function, no matter what.

Reply:
I am not sure. From outside `handleSession` should behave the same. Before `handleSession` pushed messages into `server.messages` (just early) and now it also does that (just later).

So not sure what to change here?

Reply:
@mitar Maybe just provide more context about this behavior in tests? But as I see it is done in current tests)